client/src/snippets.json

```diff
@@ -24,17 +38,114 @@
             "        $0",
             "    </when>",
-            "    <when value=\"$5\">",
+            "    <when value=\"$6\">",
             "    </when>",
             "</conditional>"
         ],
         "description": "Conditional parameter select with two options"
+    },
+    "Select parameter with 3 options": {
+        "prefix": "gx-select-3-param",
+        "body": [
+            "<param name=\"$1\" type=\"select\" label=\"$2\" help=\"$3\">",
+            "    <option value=\"$4\">TODO: option $4</option>",
+            "    <option value=\"$5\">TODO: option $5</option>",
+            "    <option value=\"$6\">TODO: option $6</option>",
+            "</param>"
+        ],
+        "description": "Select parameter with 3 options"
+    },
+    "Select parameter with 5 options": {
+        "prefix": "gx-select-5-param",
+        "body": [
+            "<param name=\"$1\" type=\"select\" label=\"$2\" help=\"$3\">",
+            "    <option value=\"$4\">TODO: option $4</option>",
+            "    <option value=\"$5\">TODO: option $5</option>",
+            "    <option value=\"$6\">TODO: option $6</option>",
+            "    <option value=\"$7\">TODO: option $7</option>",
+            "    <option value=\"$8\">TODO: option $8</option>",
+            "</param>"
+        ],
+        "description": "Select parameter with 5 options"
+    },
+    "Boolean parameter": {
+        "prefix": "gx-boolean-param",
+        "body": [
+            "<param argument=\"$1\" type=\"boolean\" truevalue=\"$2\" falsevalue=\"\" label=\"$3\" help=\"$4\" />"
+        ],
+        "description": "Boolean parameter"
+    },
+    "Integer parameter": {
+        "prefix": "gx-integer-param",
+        "body": [
+            "<param argument=\"$1\" type=\"integer\" min=\"$2\" max=\"$3\" value=\"$4\" label=\"$5\" help=\"$6\" />"
+        ],
+        "description": "Integer parameter"
+    },
+    "Float parameter": {
+        "prefix": "gx-float-param",
+        "body": [
+            "<param argument=\"$1\" type=\"float\" min=\"$2\" max=\"$3\" value=\"$4\" label=\"$5\" help=\"$6\" />"
+        ],
+        "description": "Float parameter"
+    },
+    "Data parameter with single required dataset": {
+        "prefix": "gx-data-single-param",
+        "body": [
+            "<param argument=\"$1\" name==\"$1\" type=\"data\" format==\"$2\" label=\"$3\" help=\"$4\" />"
+        ],
+        "description": "Data parameter with single required dataset"
+    },
+    "Data parameter with multiple required datasets": {
+        "prefix": "gx-data-multiple-param",
+        "body": [
+            "<param argument=\"$1\" name==\"$1\" type=\"data\" format==\"$2\" multiple=\"true\" label=\"$3\" help=\"$4\" />"
+        ],
+        "description": "Data parameter with multiple required datasets"
+    },
+    "Text parameter": {
+        "prefix": "gx-text-param",
+        "body": [
+            "<param argument=\"$1\" type=\"text\" value=\"\" label=\"$2\" help=\"$3\">",
```

Comment:
```suggestion
            "<param argument=\"$1\" type=\"text\" value=\"$2\" label=\"$3\" help=\"$4\">",
```

Reply:
The value can and very often is empty. How does the editor act here, do we need to fill this field when we add $2 and can the user just jump over?

Reply:
The user can just hit `Tab` again to jump to the next place-holder and leave it empty.